code/modules/fishing/sources/_fish_source.dm

```diff
@@ -404,3 +407,22 @@ GLOBAL_LIST(fishing_property_cache)
 			reward.pixel_y = rand(-9, 9)
 		if(severity >= EXPLODE_DEVASTATE)
 			reward.ex_act(EXPLODE_LIGHT)
+
+///Called when releasing a fish in a fishing spot with the TRAIT_CATCH_AND_RELEASE trait.
+/datum/fish_source/proc/readd_fish(obj/item/fish/fish, mob/living/releaser)
+	var/is_morbid = HAS_MIND_TRAIT(releaser, TRAIT_MORBID)
+	var/is_naive = HAS_MIND_TRAIT(releaser, TRAIT_NAIVE)
+	if(fish.status == FISH_DEAD) //ded fish won't repopulate the sea.
+		if(is_naive || is_morbid)
+			releaser.add_mood_event("fish_released", /datum/mood_event/fish_released, is_morbid && !is_naive, fish)
+		return
+	var/fish_in_table = (fish.type in fish_table)
+	if(fish_in_table != is_morbid || is_naive)
+		releaser.add_mood_event("fish_released", /datum/mood_event/fish_released, is_morbid && !is_naive, fish)
+	if(!(fish.type in fish_counts)) //This fish can be caught indefinitely so it won't matter.
+		return
```

Comment:
A: shouldn't use in for assoc lists see other pr (N^2 harms clarity slightly)
B: why the unique var